ts/components/menu/items/RetrySend/RetrySendMenuItem.tsx

```diff
@@ -0,0 +1,25 @@
+import { useCallback } from 'react';
+import { WithMessageId } from '../../../../session/types/with';
+import { useMessageDirection, useMessageStatus } from '../../../../state/selectors';
+import { ItemWithDataTestId } from '../MenuItemWithDataTestId';
+import { Data } from '../../../../data/data';
+
+export const RetryItem = ({ messageId }: WithMessageId) => {
+  const direction = useMessageDirection(messageId);
+
+  const status = useMessageStatus(messageId);
+  const isOutgoing = direction === 'outgoing';
+
+  const showRetry = status === 'error' && isOutgoing;
+  const onRetry = useCallback(async () => {
```

Comment:
Can you not use useAsync here?

Reply:
no, because we need the function itself for the onClick() below, and useAsync/useAsyncFn  return the result https://github.com/streamich/react-use/blob/master/docs/useAsyncFn.md

Reply:
useAsyncFn also outputs the fetch method as `doFetch()` which you can then call in the button. I did this as part of the network page. This should work in theory but as you want.

```tsx
 const [ ,doFetch] = useAsyncFn(async () => {
    const found = await Data.getMessageById(messageId);
    if (found) {
      await found.retrySend();
    }
  }, [messageId]);

  return showRetry ? (
    // eslint-disable-next-line @typescript-eslint/no-misused-promises
    <ItemWithDataTestId onClick={() => doFetch()}>{window.i18n('resend')}</ItemWithDataTestId>
  ) : null;
};
```

Reply:
very nice. Added